crawl-ref/source/beam.cc

```diff
@@ -4244,8 +4244,12 @@ void bolt::handle_stop_attack_prompt(monster* mon)
     else if (flavour == BEAM_CHARM)
     {
         string verb = make_stringf("charm %s", mon->name(DESC_THE).c_str());
-        if (rude_stop_summoning_prompt(verb))
-        {
+
+        bool stop = mon->res_poison() <= 0
```

Comment:
When dealing with prompts for user-visible info, we should use monster info rather than the raw monster values.